react-native-git-upgrade/checks.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+import {execSync} from 'child_process';
+import semver from 'semver';
+
+export function checkDeclaredVersion({declaredVersion}) {
+  if (!declaredVersion) {
+    throw new Error(
+      'Your \'package.json\' file doesn\'t seem to have \'react-native\' as a dependency.'
+    );
+  }
+}
+
+export function checkMatchingVersions({currentVersion, declaredVersion}) {
+  if (!semver.satisfies(currentVersion, declaredVersion)) {
+    throw new Error(
+      'react-native version in \'package.json\' doesn\'t match the installed version in \'node_modules\'.\n' +
+      'Try running \'npm install\' to fix the issue.'
```

Comment:
nit: I wonder if we should start mentioning yarn in error messages. "Try running 'npm install' or yarn to fix the issue"?